src/sonic-yang-models/yang-models/sonic-copp.yang

```diff
@@ -0,0 +1,243 @@
+					type string {
+						pattern "(sr_tcm)|(tr_tcm)|(storm)" {
+							error-message "Invalid value passed for mode";
+						}
+					}
+					description
+						"Policer mode";
+				}
+				
+				leaf color {
+					type string {
+						pattern "(aware)|(blind)" {
+							error-message "Invalid value passed for color";
+						}
+					}
+					description
+						"Policer color mode";
+				}
+
+				leaf cir {
+					type uint64;
+					description
+						"Committed information rate for the dual-rate token
+						bucket policer.  This value represents the rate at which 
+						tokens are added to the primary bucket.";
+				}
+
+				leaf cbs {
+					type uint64;
+					description
+						"Committed burst size for the dual-rate token bucket
+						policer.  This value represents the depth of the token
+						bucket.";
+
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "cbs can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "cbs must be greater than or equal to cir";
+					}
+				}
+
+				leaf pir {
+					type uint64;
+					description
+						"Peak information rate for the dual-rate token bucket
+						policer.  This value represents the rate at which tokens
+						are added to the secondary bucket.";
+
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "pir can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "pir must be greater than or equal to cir";
+					}
+				}
+
+				leaf pbs {
+					type uint64;
+					description
+						"Excess burst size for the dual-rate token bucket policer.
+						This value represents the depth of the secondary bucket.";
+
+					must "((current()/../pir) and (current()/../pir > 0))" {
+						error-message "pbs can't be configured without pir.";
+					}
+					must "(current() >= current()/../pir)" {
+						error-message "pbs must be greater than or equal to pir";
+					}
+					must "((not(current()/../cbs)) or (current() >= current()/../cbs))" {
+						error-message "pbs must be greater than or equal to cbs";
+					}
+				}
+				
+				leaf green_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for green_action";
+						}
+					}
+					description
+						"Green action";
+				}
+				
+				leaf yellow_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for yellow_action";
+						}
+					}
+					description
+						"Yellow action";
+				}
+
+				leaf red_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for red_action";
+						}
+					}
+					description
+						"Red action";
+				}
+				
+				leaf genetlink_name {
```

Comment:
If genetlink name and mc_group name are exposed to north bound interfaces to be modified what is the purpose of restricting here to a single value?

Reply:
genetlink_name and genetlink_mcgrp_name are applicable to sFlow only. Current supported values are "psample" and "packets" respectively.  Default [copp_cfg.j2](https://github.com/Azure/sonic-buildimage/blob/master/files/image_config/copp/copp_cfg.j2#L54) does have these defined. However, user may chose to create a new copp_group  COPP_GROUP|sflow-group and bind it to COPP_TRAP|sflow. Hence these fields are defined in yang and restricted to allowed patterns (psample, packets). 
```
genetlink_name       = genetlink_name ;[Optional] "psample" for sFlow 
genetlink_mcgrp_name = multicast group name; ;[Optional] "packets" for sFlow 
```
@dgsudharsan, Are you suggesting not to expose these fields through yang ?

Reply:
We do not expose these to user as these are internal implementations. I am not sure why an external user would be interested in specifying details about internal communication mechanisms. @prsunny Please let me your thoughts on this.

Reply:
Agree, I think genetlink part can be removed from Yang and it is internal.